src/model_config_utils.cc

```diff
@@ -233,10 +239,31 @@ ValidateEnsembleSchedulingConfig(const inference::ModelConfig& config)
                                          "' is not used");
     }
     if (!it->second.ready) {
-      return Status(
-          Status::Code::INVALID_ARG, "output '" + output.name() +
-                                         "' for ensemble '" + config.name() +
-                                         "' is not written");
+      std::string error_message = "output '" + output.name() +
+                                  "' for ensemble '" + config.name() +
+                                  "' is not written";
+
+      // recurrsively check 'prev_nodes' for the source of not-ready state
```

Comment:
Is there a possibility of being stuck in a cycle during this check?

Reply:
Yes and fixed.